### ISSUE TITLE
State and address

### ISSUE THREAD

**Issue by EduardoRFS** [marigold-dev/deku#294]

**Comment by EduardoRFS** [comment]
@anchpop what is the format of Tezos contract address?

**Comment by anchpop** [comment]
Not sure yet - posting this to remind myself to look into `Contract.fresh_contract_from_current_nonce` tomorrow

**Comment by anchpop** [comment]
@EduardoRFS I think they just [hash the nonce](https://github.com/marigold-dev/deku/blob/cee5b19d6c7e9b7e35a0c363b4e10d65c3060ec5/ligolang/vendors/tezos-ligo/src/proto_alpha/lib_protocol/contract_repr.ml#L154)

**Comment by EduardoRFS** [comment]
@anchpop can you write down what we discussed and what will be the chosen solution?

**Comment by anchpop** [comment]
Yes, the address will the the hash of the operation that originated the contract, or the hash of the hash
